common/src/main/java/org/conscrypt/OpenSSLSocketImpl.java

```diff
@@ -16,1299 +16,118 @@
  */
 @Internal
-public class OpenSSLSocketImpl
-        extends javax.net.ssl.SSLSocket
-        implements NativeCrypto.SSLHandshakeCallbacks, SSLParametersImpl.AliasChooser,
-        SSLParametersImpl.PSKCallbacks {
-
-    private static final boolean DBG_STATE = false;
-
-    /**
-     * Protects handshakeStarted and handshakeCompleted.
-     */
-    private final Object stateLock = new Object();
-
-
-    // @GuardedBy("stateLock");
-    private int state = STATE_NEW;
-
-    /**
-     * Protected by synchronizing on stateLock. Starts as 0, set by
-     * startHandshake, reset to 0 on close.
-     */
-    // @GuardedBy("stateLock");
-    private long sslNativePointer;
-
-    /**
-     * Protected by synchronizing on stateLock. Starts as null, set by
-     * getInputStream.
-     */
-    // @GuardedBy("stateLock");
-    private SSLInputStream is;
-
-    /**
-     * Protected by synchronizing on stateLock. Starts as null, set by
-     * getInputStream.
-     */
-    // @GuardedBy("stateLock");
-    private SSLOutputStream os;
-
-    private final Socket socket;
-    private final boolean autoClose;
-
-    /**
-     * The peer's DNS hostname if it was supplied during creation. Note that
-     * this may be a raw IP address, so it should be checked before use with
-     * extensions that don't use it like Server Name Indication (SNI).
-     */
-    private String peerHostname;
-
-    /**
-     * The peer's port if it was supplied during creation. Should only be set if
-     * {@link #peerHostname} is also set.
-     */
-    private final int peerPort;
-
-    private final SSLParametersImpl sslParameters;
-
-    /*
-     * A CloseGuard object on Android. On other platforms, this is nothing.
-     */
-    private final Object guard = Platform.closeGuardGet();
-
-    private ArrayList<HandshakeCompletedListener> listeners;
-
-    /**
-     * Private key for the TLS Channel ID extension. This field is client-side
-     * only. Set during startHandshake.
-     */
-    private OpenSSLKey channelIdPrivateKey;
-
-    /** Set during startHandshake. */
-    private AbstractOpenSSLSession sslSession;
-
-    /** Used during handshake callbacks. */
-    private AbstractOpenSSLSession handshakeSession;
-
-    /**
-     * Local cache of timeout to avoid getsockopt on every read and
-     * write for non-wrapped sockets. Note that
-     * OpenSSLSocketImplWrapper overrides setSoTimeout and
-     * getSoTimeout to delegate to the wrapped socket.
-     */
-    private int readTimeoutMilliseconds = 0;
-    private int writeTimeoutMilliseconds = 0;
-
-    private int handshakeTimeoutMilliseconds = -1;  // -1 = same as timeout; 0 = infinite
-
-    OpenSSLSocketImpl(SSLParametersImpl sslParameters) throws IOException {
-        this.socket = this;
-        this.peerHostname = null;
-        this.peerPort = -1;
-        this.autoClose = false;
-        this.sslParameters = sslParameters;
+public abstract class OpenSSLSocketImpl extends AbstractConscryptSocket {
```

Comment:
Why are all the constructors back to public? I don't think this will help anyone. If they used this class, it was via `OpenSSLSocketFactoryImpl` to create it and then casting to this class or using reflection to call a certain method on the instance returned.